PWGCF/TwoParticleCorrelations/Tasks/lambdaR2Correlation.cxx

```diff
@@ -1136,74 +1101,67 @@ struct lambdaCorrelationAnalysis {
     int phibin1 = static_cast<int>(p1.phi() / phibinwidth);
     int phibin2 = static_cast<int>(p2.phi() / phibinwidth);
 
-    float eff_1 = 1., eff_2 = 1.;
-
-    if constexpr (rec_gen == kRec) {
-      if (part_pair == kLambdaAntiLambda) {
-        get_corr_factor<kLambda>(eff_1, p1.pt());
-        get_corr_factor<kAntiLambda>(eff_2, p2.pt());
-      } else if (part_pair == kLambdaLambda) {
-        get_corr_factor<kLambda>(eff_1, p1.pt());
-        get_corr_factor<kLambda>(eff_2, p2.pt());
-      } else if (part_pair == kAntiLambdaAntiLambda) {
-        get_corr_factor<kAntiLambda>(eff_1, p1.pt());
-        get_corr_factor<kAntiLambda>(eff_2, p2.pt());
-      }
-    }
+    float corfac1 = 1., corfac2 = 1.;
+
+    get_corr_factor<rec_gen>(corfac1, col, p1);
+    get_corr_factor<rec_gen>(corfac2, col, p2);
```

Comment:
This is highly inefficient
Each track correction factor is search a number of times equal to the multiplicity
If the multiplicity is only the number of Lambdas then the impact will be small
But if there are execution time issues definitely this would need to be modified

Reply:
Dear Victor,
I understand this. I will try to add these correction factors in the table itself while producing tables. Would that be efficient ?

Reply:
Initially, that sound a good approach